docs/how-to-guides/develop-a-prompty/prompty-output-format.md

```diff
@@ -0,0 +1,230 @@
+    "answer": "The meaning of life is a philosophical question that varies depending on individual beliefs and perspectives."
+}
+```
+
+Users can also specify the returned fields by configuring output.
+```yaml
+---
+name: Json Format Prompt
+description: A basic prompt that uses the GPT-3 chat API to answer questions
+model:
+    api: chat
+    configuration:
+      type: azure_openai
+      azure_deployment: gpt-35-turbo-0125
+      connection: open_ai_connection
+    parameters:
+      max_tokens: 128
+      temperature: 0.2
+      response_format:
+        type: json_object
+sample:
+  inputs:
+    first_name: John
+    last_name: Doh
+    question: what is the meaning of life?
+  outputs:
+    answer: The meaning of life is a philosophical question
+---
+system:
+You are an AI assistant who helps people find information.
+As the assistant, you answer questions briefly, succinctly. Your structured response. Only accepts JSON format, likes below:
+{"name": customer_name, "answer": the answer content}
+
+# Customer
+You are helping {{first_name}} {{last_name}} to find answers to their questions.
+Use their name to address them in your responses.
+
+user:
+{{question}}
+```
+Prompty will eventually return the user-specified outputs.
+```json
+{
+  "answer": "The meaning of life is a philosophical question that varies depending on individual beliefs and perspectives."
+}
+```
+
+### All choices
+In some scenarios, users may need to obtain the original response of llm to perform some operations. It can be obtained by configuring `response=all` to get the original [LLM response](https://platform.openai.com/docs/api-reference/chat/object).
+
+```yaml
+---
+name: All Choices Text Format Prompt
+description: A basic prompt that uses the GPT-3 chat API to answer questions
+model:
+    api: chat
+    configuration:
+      type: azure_openai
+      connection: open_ai_connection
+      azure_deployment: gpt-35-turbo-0125
+    parameters:
+      max_tokens: 128
+      temperature: 0.2
+      n: 3
+    response: all
+sample:
+  inputs:
+    first_name: John
+    last_name: Doh
+    question: what is the meaning of life?
+---
+system:
+You are an AI assistant who helps people find information.
+As the assistant, you answer questions briefly, succinctly,
+and in a personable manner using markdown and even add some personal flair with appropriate emojis.
+
+# Safety
+- You **should always** reference factual statements to search results based on [relevant documents]
+- Search results based on [relevant documents] may be incomplete or irrelevant. You do not make assumptions
+# Customer
+You are helping {{first_name}} {{last_name}} to find answers to their questions.
+Use their name to address them in your responses.
+
+user:
+{{question}}
+```
+
+### Streaming output
+When `stream=true` is configured in the parameters of a prompty whose output format is text, promptflow sdk will return a generator type, which item is the content of each chunk.
```

Comment:
for the last sentence, might be "which yields each chunk"? - maybe let GPT help on this.

Reply:
refine it by copilot, "Each item from the generator represents the content of a chunk."